pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/ClassTypeResolverTest.java

```diff
@@ -659,7 +660,117 @@ public void testSuperExpression() throws JaxenException {
         assertEquals("All expressions not tested", index, expressions.size());
     }
 
+    @Test
+    public void testFieldAccess() throws JaxenException {
```

Comment:
I'd consider splitting this into a few separate tests, so if something breaks we may know instead of just stopping at the first failure. I suggest we take each scenario apart:

 - "normal" field access
 - "shadowed" field access
 - "shadowed by inheritance" field access
 - "super" field access
 - so on

This would also make tests more readable, and in a way, force you to keep the test classes smaller / neater, and therefore easier to maintain making sure there are no screwups. This is most important since the test classes are completely meaningless code, and therefore, screwups are easy and hard to detect.